src/notification/subscribe/subscribe.test.ts

```diff
@@ -0,0 +1,105 @@
+
+  beforeEach(() => {
+    app = express();
+
+    const subscribeRouter = express.Router();
+
+    subscribeRouter.get('/status', (req, res) => {
+      res.status(200).json({ message: 'Subscriber is running' });
+    });
+
+    app.use('/subscribe', subscribeRouter);
+
+    // Ensure that the mock client is set up correctly before each test
+    client = mockClient();
+    (mqtt.connect as jest.Mock).mockReturnValue(client);
+    (connectToMongoDB as jest.Mock).mockResolvedValue(undefined);
+
+    // Ensure MQTT client setup is correctly mocked
+    const connectMock = mqtt.connect as jest.Mock;
+    console.log('connectMock call count:', connectMock.mock.calls.length); // Debugging: Check the number of connect calls
+
+    const mockClientInstance = connectMock.mock.results[0]?.value as Partial<MqttClient>;
+    if (mockClientInstance) {
+      (mockClientInstance.on as jest.Mock).mockImplementation((event: string, handler: (...args: any[]) => void) => {
+        if (event === 'message') {
+          console.log('Mock on handler triggered'); // Debugging: Ensure this line is executed
+          handler('test-topic', Buffer.from('test-message'));
+        }
+        return mockClientInstance as MqttClient;
+      });
+    }
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return 200 and a status message when /subscribe/status is called', async () => {
+    const response = await request(app).get('/subscribe/status');
+
+    expect(response.status).toBe(200);
+    expect(response.body).toEqual({ message: 'Subscriber is running' });
+  });
+
+  it('should log messages received from the broker', () => {
+    const logSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
+
+    const connectMock = jest.fn();
+    connectMock.mockReturnValueOnce('Received message from topic test-topic: test-message');
+    console.log('connectMock call count after setup:', connectMock.mock.calls.length); // Debugging: Check the number of connect calls
+
+    // FIXME: mockClientInstance returns an undefined value 
+    // const mockClientInstance = connectMock.mock.results[0]?.value as Partial<MqttClient>;
+
+    // if (mockClientInstance && mockClientInstance.on) {
+    //   console.log('mockClientInstance.on:', mockClientInstance.on); // Debugging: Check the on method
+
+    //   // Directly trigger the 'message' event handler
+    //   const messageHandler = (mockClientInstance.on as jest.Mock).mock.calls.find(([event]) => event === 'message')?.[1];
+    //   if (messageHandler) {
+    //     messageHandler('test-topic', Buffer.from('test-message')); // Trigger the message event
+    //   }
+    // }
+
+    const expectedMessage = 'Received message from topic test-topic: test-message';
+
+    // FIXME: This code returns [["connectMock call count after setup:", 0], ["Actual calls to log:", [Circular]]]
+    // const actualCalls = logSpy.mock.calls.map((call: any[]) => call.join(' '));
+    // logSpy.mockRestore();
```

Comment:
**question:** is this commented out code still needed?  If not, could it be removed?

Reply:
resolved: this can be used at later time